### PR TITLE
Layer tus-js-client on top of Axios

### PR DESCRIPTION
**Pull request by SpecLad**
### Motivation and context

This ensures that requests coming from tus-js-client have the same defaults as the ones coming from the rest of the UI. In particular, this ensures that TUS requests include the `X-CSRFTOKEN` header.

Currently, this doesn't matter much, because TUS requests are authenticated using the token. However, I'd like to get rid of token authentication in the UI, after which `X-CSRFTOKEN` will become important.

### How has this been tested?

Manual testing.

### Checklist

- [x] I submit my changes into the `develop` branch
- ~~[ ] I have created a changelog fragment~~ 
- ~~[ ] I have updated the documentation accordingly~~
- ~~[ ] I have added tests to cover my changes~~
- ~~[ ] I have linked related issues (see [GitHub docs](
  https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword))~~
- [x] I have increased versions of npm packages if it is necessary
  ([cvat-canvas](https://github.com/cvat-ai/cvat/tree/develop/cvat-canvas#versioning),
  [cvat-core](https://github.com/cvat-ai/cvat/tree/develop/cvat-core#versioning),
  [cvat-data](https://github.com/cvat-ai/cvat/tree/develop/cvat-data#versioning) and
  [cvat-ui](https://github.com/cvat-ai/cvat/tree/develop/cvat-ui#versioning))

### License

- [x] I submit _my code changes_ under the same [MIT License](
  https://github.com/cvat-ai/cvat/blob/develop/LICENSE) that covers the project.
  Feel free to contact the maintainers if that's a concern.

## Summary by CodeRabbit

- **New Features**
  - Introduced a new integration for resumable uploads using Axios and the tus protocol, enhancing upload reliability and control.
  
- **Bug Fixes**
  - Updated version from 15.1.0 to 15.1.1 for minor fixes and improvements.

- **Refactor**
  - Streamlined the `chunkUpload` function by removing organization-specific parameters and callbacks, refining the upload process.

### CODE REVIEW

File: cvat-core/src/axios-tus.ts [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+    readonly #axiosResponse: AxiosResponse;
+
+    constructor(axiosResponse: AxiosResponse) {
+        this.#axiosResponse = axiosResponse;
+    }
+
+    getStatus(): number { return this.#axiosResponse.status; }
+    getHeader(header: string): string { return this.#axiosResponse.headers[header.toLowerCase()]; }
+    getBody(): string { return this.#axiosResponse.data; }
+    getUnderlyingObject(): any { return this.#axiosResponse; }
```
Comment (bsekachev) [inline cvat-core/src/axios-tus.ts:18]:
```suggestion
    getUnderlyingObject(): AxiosResponse { return this.#axiosResponse; }
```

File: cvat-core/src/axios-tus.ts [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+import * as tus from 'tus-js-client';
+
+class AxiosHttpResponse implements tus.HttpResponse {
+    readonly #axiosResponse: AxiosResponse;
+
+    constructor(axiosResponse: AxiosResponse) {
+        this.#axiosResponse = axiosResponse;
+    }
+
+    getStatus(): number { return this.#axiosResponse.status; }
```
Comment (bsekachev) [inline cvat-core/src/axios-tus.ts:15]: Please, follow the general codestyle, e.g.

```
getStatus(): number {
    return this.#axiosResponse.status;
}
```

File: cvat-core/src/axios-tus.ts [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+            handler(progressEvent.loaded);
+        };
+    }
+
+    async send(body: any): Promise<tus.HttpResponse> {
+        const axiosResponse = await Axios({ ...this.#axiosConfig, data: body });
+        return new AxiosHttpResponse(axiosResponse);
+    }
+
+    async abort(): Promise<void> { this.#abortController.abort(); }
```
Comment (bsekachev) [inline cvat-core/src/axios-tus.ts:53]:
```suggestion
    async abort(): Promise<void> {
        await this.#abortController.abort();
    }
```

File: cvat-core/src/axios-tus.ts [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+
+    getUnderlyingObject(): any { return this.#axiosConfig; }
```
Comment (bsekachev) [inline cvat-core/src/axios-tus.ts:55]:
```suggestion
    getUnderlyingObject(): AxiosRequestConfig { return this.#axiosConfig; }
```

File: cvat-core/src/axios-tus.ts [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+
+class AxiosHttpResponse implements tus.HttpResponse {
+    readonly #axiosResponse: AxiosResponse;
+
+    constructor(axiosResponse: AxiosResponse) {
+        this.#axiosResponse = axiosResponse;
+    }
+
+    getStatus(): number { return this.#axiosResponse.status; }
+    getHeader(header: string): string { return this.#axiosResponse.headers[header.toLowerCase()]; }
```
Comment (bsekachev) [inline cvat-core/src/axios-tus.ts:16]: Probably it also may return `undefined` as in `AxiosHttpRequest`

```suggestion
    getHeader(header: string): string | undefined { return this.#axiosResponse.headers[header.toLowerCase()]; }
```

File: cvat-core/src/axios-tus.ts (hunk 3 above)
Comment (bsekachev) [reply cvat-core/src/axios-tus.ts:53]: Just looked at the implementation of abort controller. It looks like the method is sync. Is there any reason to make it this wrapper method async here?

Comment (SpecLad) [reply cvat-core/src/axios-tus.ts:53]: The `abort` function is [defined](https://github.com/tus/tus-js-client/blob/main/docs/api.md#httpstack) as returning a promise (I don't really know why...), so I made it `async` to make JS return a promise implicitly.

File: cvat-core/src/axios-tus.ts (hunk 1 above)
Comment (SpecLad) [reply cvat-core/src/axios-tus.ts]: Fixed.

File: cvat-core/src/axios-tus.ts (hunk 2 above)
Comment (SpecLad) [reply cvat-core/src/axios-tus.ts]: Okay, reformatted.

File: cvat-core/src/axios-tus.ts (hunk 4 above)
Comment (SpecLad) [reply cvat-core/src/axios-tus.ts]: Fixed.

File: cvat-core/src/axios-tus.ts (hunk 5 above)
Comment (SpecLad) [reply cvat-core/src/axios-tus.ts]: Good point, fixed. This is actually a bug in the interface definition, so I also filed tus/tus-js-client#706.

**Review by bsekachev — APPROVED** [review]

**Pull request merged**